### DISCUSSION TITLE
Nuxt3: Fetch Data Serverside

### DISCUSSION THREAD

**Discussion by JanWerder** [nuxt/nuxt#18229] [Questions] [answered]
Hey,

I want to fetch data in a component and it would be best to do this serverside, so the client gets the data no matter if Javascript is enabled or not.

My code is simple so far.
```
<template>
  <div>
    <span v-for="ship of ships" :key="ship.shipName">{{ ship.longName }}</span>
  </div>
</template>
<script>
export default {
  async setup() {
    const { data: ships } = await useFetch("/api/ships", { server: true, defer: false });
    return { ships };

    // const { data: ships } = await useAsyncData('ships', () => $fetch('/api/ships'), { server: true, defer: false });
    // return { ships };

  },
};
</script>
```
(The API returns valid data. It's a Java Spring Boot Backend, that I expose via a Caddy reverse proxy, so it appears on the same ports as Vue in dev mode.)

But both approaches with either `useFetch` or `useAsyncData` only fetch the data clientside and not on the first visit of the component, only when I navigate and come back to it a few seconds later.

Where did I go wrong? Did I miss something?

**Comment by wobsoriano** [comment]
Try to log ships `ships` and check if you get the data in both your terminal and browser console.

**Comment by danielroe** [comment, marked as answer]
The issue is almost certainly that the isomorphic `$fetch` thinks that the URL (because it is the same host) can be served from within your Nitro server. (But there is no endpoint accessible within Nitro that can respond.)

Instead, you can call `useFetch` with a full URL, which will force a network request to be made.

**Reply by JanWerder** [reply]
That works, thanks. Is there anything else beside providing the full URL, which of course differs between dev & prod, to circumvent that?
